File: samples/client/3_0_3_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/MapSchemaValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;

import java.util.List;
import java.util.Map;

public interface MapSchemaValidator <OutType, BoxedType> {
    OutType getNewInstance(Map<?, ?> arg, List<Object> pathToItem, PathToSchemasMap pathToSchemas) throws ValidationException;
    OutType validate(Map<?, ?> arg, SchemaConfiguration configuration) throws ValidationException;
    BoxedType validateAndBox(Map<?, ?> arg, SchemaConfiguration configuration) throws ValidationException;
}

File: samples/client/3_1_0_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/MapSchemaValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

interface MapSchemaValidator<OutType, BoxedType> {
    @Throws(ValidationException::class)
    fun getNewInstance(arg: Map<*, *>, pathToItem: List<Any>, pathToSchemas: PathToSchemasMap): OutType

    @Throws(ValidationException::class)
    fun validate(arg: Map<*, *>, configuration: SchemaConfiguration?): OutType

    @Throws(ValidationException::class)
    fun validateAndBox(arg: Map<*, *>, configuration: SchemaConfiguration?): BoxedType
}